x-pack/legacy/plugins/apm/public/components/app/ApmIndicesPermission/index.tsx

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiButton,
+  EuiEmptyPrompt,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLink,
+  EuiPanel,
+  EuiText,
+  EuiTitle
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { isEmpty } from 'lodash';
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { FETCH_STATUS, useFetcher } from '../../../hooks/useFetcher';
+import { fontSize, pct, px, units } from '../../../style/variables';
+import { ElasticDocsLink } from '../../shared/Links/ElasticDocsLink';
+import { SetupInstructionsLink } from '../../shared/Links/SetupInstructionsLink';
+
+export const APMIndicesPermission: React.FC = ({ children }) => {
+  const [
+    isPermissionWarningDismissed,
+    setIsPermissionWarningDismissed
+  ] = useState(false);
+
+  const { data: indicesPrivileges = {}, status } = useFetcher(callApmApi => {
+    return callApmApi({
+      pathname: '/api/apm/security/indices_privileges'
+    });
+  }, []);
+
+  // Return null until receive the reponse of the api.
+  if (status === FETCH_STATUS.LOADING || status === FETCH_STATUS.PENDING) {
+    return null;
+  }
+
+  const indicesPrivilegesKeys = Object.keys(indicesPrivileges);
+  const indicesWithoutPermission = indicesPrivilegesKeys.filter(
+    index => !indicesPrivileges[index].read
+  );
+
+  // Show permission warning when a user has at least one index without Read privilege,
+  // and he has not manually dismissed the warning
+  if (!isEmpty(indicesWithoutPermission) && !isPermissionWarningDismissed) {
+    return (
+      <PermissionWarning
+        indicesWithoutPermission={indicesWithoutPermission}
+        // Show escape hatch button if the index pattern was used to check the privileges
+        // or if at least one index has read privilege
+        showEscapeHatch={
+          indicesPrivilegesKeys.some(index => index === 'apm-*') ||
```

Comment:
@sqren I'm not sure if I should check the `apm-*` hardcoded like that. Can I get it somewhere?

Reply:
I don't think hardcode anything is necessary. Is it not possible to list the indices (regardless what they are) if the user can't access them?

Reply:
I'm doing it to show the Dismiss button. So I'd show the button if I have `apm-*` in the list (because in that case, the user might have access to other indices) or if the list of privileges (returned by the API) is different than the list of indices without privileges.

Reply:
I'm don't think we should treat `apm-*` any different than `apm*`, `ap*` or `apm-*-transactions-*`.

If one or more APM indices are configured to be a particular range of indicies (eg `apm-*`), the user should also have a read access to that same range. 

I can still see how someone can misconfigure it though. Eg. leaving the apm indices as the default (`apm-*`) while only allowing access on a more granular level (`apm-*-transaction-*`, `apm-*-span-*` etc). 
So I think we should just always show the dismiss button.